lib/routes/objects.js

```diff
@@ -109,13 +111,14 @@ module.exports =
               .filter((variant) => variant !== '')
             : null;
 
-          if (!variants) {
-            const allObjVariants = await fastify.getAllObjectVariants({
+          if (!variants && allObjVariants) {
+            return allObjVariants;
+          } else if (!variants) {
+            return fastify.getAllObjectVariants({
```

Comment:
I might be missing something, but this clause seems odd. The block above calls `getAllObjectVariants` already and returns it if `!variants`. Under what circumstance would calling `getAllObjectVariants` again here help?

Reply:
Yeah it's a bit confusing but it is correct. We need to return `allObjVariants` if `variants` are not specified. That said, if in the step before I have already fetched `allObjVariants` to check on the `version`, then I can return these. Otherwise if the version was not specified and therefore we did not fetch `allObjVariants`, we need to fetch `allObjVariants` before returning.

Reply:
It's an optimization. It's to avoid to fetch twice allObjVariants if we already fetched in the step before.

Reply:
Additionally we use `getAllObjectVariants` instead of `getAllObjectVersions` to verify a version exists.

Reply:
Ah, that makes sense. Thanks for clarifying.